src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -42,38 +42,32 @@ public EventConsumer(MeterRegistry registry) {
     @Acknowledgment(Strategy.PRE_PROCESSING)
     // Can be modified to use Multi<Message<String>> input also for more concurrency
     public Uni<Void> processAsync(Message<String> input) {
-        if (log.isLoggable(Level.FINE)) {
-            log.fine("Processing: " + input.getPayload());
-        }
-        return Uni.createFrom()
-                .item(input)
+        return Uni.createFrom().item(() -> input.getPayload())
+                .onItem().invoke(payload -> log.fine(() -> "Processing: " + payload))
```

Comment:
This seems to lose the `if (isLoggable())` and is thus always logging at fine level, which is not desirable.

Reply:
The `"Processing: " + payload` concatenation won't be performed before the logger checks if fine is loggable (it is done internally) because there's a supplier now: `() -> "Processing: " + payload)`.

Reply:
Ok. I was less worried about the concatenation, but more about unmarshalling a potentially large payload to string and then throwing it away. Great, learned a new thing.

Reply:
Most loggers (if not all) check internally if a level is loggable before doing it. But the concatenation part is important: it will be done before the internal check is done unless we use a Supplier.